crates/re_analytics/src/native/pipeline.rs

```diff
@@ -197,36 +218,43 @@ fn realtime_pipeline(
     tick: Duration,
     event_tx: &channel::Sender<Result<Event, RecvError>>,
     event_rx: &channel::Receiver<Result<Event, RecvError>>,
-) -> anyhow::Result<()> {
-    let analytics_id = config.analytics_id.clone();
-    let session_id = config.session_id.to_string();
+    abort_signal: &AbortSignal,
+) {
+    let analytics_id: Arc<str> = config.analytics_id.clone().into();
+    let session_id: Arc<str> = config.session_id.to_string().into();
     let is_first_run = config.is_first_run();
 
     let ticker_rx = crossbeam::channel::tick(tick);
 
     let on_tick = |session_file: &mut _, _elapsed| {
+        // A number of things can fail here, in all cases we will stop retrying.
+        // The next time the analytics boots up, the catchup thread should handle
+        // any remaining events.
+
         if is_first_run {
             // We never send data on first run, to give end users an opportunity to opt-out.
-            return;
+            return abort_signal.signal();
         }
 
-        if let Err(err) = flush_events(session_file, &analytics_id, &session_id, sink) {
+        if let Err(err) = flush_events(session_file, &analytics_id, &session_id, sink, abort_signal)
+        {
             re_log::debug_once!("couldn't flush analytics data file: {err}");
             // We couldn't flush the session file: keep it intact so that we can retry later.
-            return;
+            return abort_signal.signal();
         }
 
         if let Err(err) = session_file.set_len(0) {
             re_log::debug_once!("couldn't truncate analytics data file: {err}");
             // We couldn't truncate the session file: we'll have to keep it intact for now, which
             // will result in duplicated data that we'll be able to deduplicate at query time.
-            return;
+            return abort_signal.signal();
         }
         if let Err(err) = session_file.rewind() {
             // We couldn't reset the session file… That one is a bit messy and will likely break
             // analytics for the entire duration of this session, but that really _really_ should
             // never happen.
             re_log::debug_once!("couldn't seek into analytics data file: {err}");
+            abort_signal.signal();
```

Comment:
`return` here?

Reply:
clippy complains about it being unneeded, i'll just silence it